client/client-multi/client-c/src/kaa/kaa_user.c

```diff
@@ -193,6 +234,77 @@ kaa_error_t kaa_user_manager_attach_to_user(kaa_user_manager_t *self
     return KAA_ERR_NONE;
 }
 
+kaa_error_t kaa_user_manager_attach_endpoint(kaa_user_manager_t *self, const char *endpoint_access_token, kaa_endpoint_status_listener_t *listener)
+{
+    KAA_RETURN_IF_NIL2(self, endpoint_access_token, KAA_ERR_BADPARAM);
+
+    KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Going to attach endpoint by access token "
+                                              "(endpoint_access_token = \"%s\")"
+                                              , endpoint_access_token);
+
+    kaa_endpoint_info_t *info = KAA_CALLOC(1, sizeof(kaa_endpoint_info_t));
+    KAA_RETURN_IF_NIL(info, KAA_ERR_NOMEM);
+
+
+    info->access_token_length = strlen(endpoint_access_token);
+    info->access_token        = KAA_MALLOC(info->access_token_length);
+
+    if (!info->access_token) {
+        dtor_endpoint_info((void*)info);
+        return KAA_ERR_NOMEM;
+    }
+
+    memcpy(info->access_token, endpoint_access_token, info->access_token_length);
+
+    if (listener)
+        info->listener = listener;
+
+    info->request_id = ++self->endpoint_request_counter;
+
+    if (!kaa_list_push_back(self->attach_endpoints, (void*)info)) {
+        dtor_endpoint_info((void*)info);
+        return KAA_ERR_NOMEM;
+    }
+
+    kaa_transport_channel_interface_t *channel =
+            kaa_channel_manager_get_transport_channel(self->channel_manager, user_sync_services[0]);
+    if (channel)
+        channel->sync_handler(channel->context, user_sync_services, 1);
+
+    return KAA_ERR_NONE;
+}
+
+kaa_error_t kaa_user_manager_detach_endpoint(kaa_user_manager_t *self, const kaa_endpoint_id_p endpoint_hash_key, kaa_endpoint_status_listener_t *listener)
+{
+    KAA_RETURN_IF_NIL2(self, endpoint_hash_key, KAA_ERR_BADPARAM);
+
+    KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Detach endpoint");
```

Comment:
1. "Going to detach endpoint" is better. 
2. Please, Use INFO level because this is an important runtime event.